Source/engine/palette.cpp

```diff
@@ -284,30 +267,37 @@ void PaletteFadeIn(int fr, const std::array<SDL_Color, 256> &srcPalette)
 	if (demo::IsRunning())
 		fr = 0;
 
-	ApplyToneMapping(logical_palette, srcPalette);
+	SDL_Color palette[256];
+	ApplyGlobalBrightness(palette, srcPalette.data());
 
 	if (fr > 0) {
 		const uint32_t tc = SDL_GetTicks();
 		fr *= 3;
 		uint32_t prevFadeValue = 255;
 		for (uint32_t i = 0; i < 256; i = fr * (SDL_GetTicks() - tc) / 50) {
-			if (i != prevFadeValue) {
-				// We can skip hardware cursor update for fade level 0 (everything is black).
-				SetFadeLevel(i, /*updateHardwareCursor=*/i != 0u, logical_palette);
-				prevFadeValue = i;
+			if (i == prevFadeValue) continue;

```

Comment:
Feels like we should add `SDL_Delay()` here.

Reply:
Done. I've added `SDL_Delay(1)`.